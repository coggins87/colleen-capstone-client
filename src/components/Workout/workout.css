.saved_workout_head {
  margin-bottom: 0;
}

.saved_workout_item {
  padding: 5px;
}