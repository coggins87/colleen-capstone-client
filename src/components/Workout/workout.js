import React from "react";
import uuid from "uuid";
import "./workout.css";

export default class Workout extends React.Component {
  
  
  render() {
    return ( <div className="workout">
      <ul>
        <h3 className="saved_workout_head">
          In {this.props.workout.workout_length} minutes:
        </h3>
        {this.props.workout.movements.map(movement => {
          return (
            <li className="saved_workout_item" key={uuid()}>
              {movement.reps} {movement.movement_name}{" "}
              {movement.equipment ? `with ${movement.equipment}` : ""}
            </li>
          );
        })}
      </ul>
      <button className="delete_button" onClick={()=>this.props.handleDelete(this.props.userId, this.props.workout.workout_id)}>Delete Workout</button>
      </div>
    );
  }
}
