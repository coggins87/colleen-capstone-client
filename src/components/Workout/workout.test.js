import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Workout from './workout'

jest.mock('uuid', () => jest.fn(() => '1'));
describe(`Workout component`, () => {
  const props = {
  
    workout: {workout_length: 1, movements:[{movement_name: 'name'},{reps: 'reps'}, {equipment: 'equip'}]}
  }



  it('renders the Workout given props', () => {
    const wrapper = shallow(<Workout {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})