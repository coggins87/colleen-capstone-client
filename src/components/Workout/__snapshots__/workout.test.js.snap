// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workout component renders the Workout given props 1`] = `
<div
  className="workout"
>
  <ul>
    <h3
      className="saved_workout_head"
    >
      In 
      1
       minutes:
    </h3>
    <li
      className="saved_workout_item"
      key="1"
    >
       
      name
       
    </li>
    <li
      className="saved_workout_item"
      key="1"
    >
      reps
       
       
    </li>
    <li
      className="saved_workout_item"
      key="1"
    >
       
       
      with equip
    </li>
  </ul>
  <button
    className="delete_button"
    onClick={[Function]}
  >
    Delete Workout
  </button>
</div>
`;
